cumulusci/tasks/salesforce/sobject_preflights.py

```diff
@@ -0,0 +1,78 @@
+from cumulusci.tasks.salesforce import BaseSalesforceApiTask
+from cumulusci.core.exceptions import TaskOptionsError
+
+from simple_salesforce.exceptions import SalesforceMalformedRequest
+
+
+class CheckSobjectOWDs(BaseSalesforceApiTask):
+    task_options = {
+        "org_wide_defaults": {
+            "description": "The Organization-Wide Defaults to check, "
+            "organized as a list with each element containing the keys api_name, "
+            "internal_sharing_model, and external_sharing_model. NOTE: you must have "
+            "External Sharing Model turned on in Sharing Settings to use the latter feature. "
+            "Checking External Sharing Model when it is turned off will fail the preflight.",
+            "required": True,
+        }
+    }
+
+    def _init_options(self, kwargs):
+        super()._init_options(kwargs)
+
+        if "org_wide_defaults" not in self.options:
+            raise TaskOptionsError("org_wide_defaults is a required option")
+
+        if not all("api_name" in entry for entry in self.options["org_wide_defaults"]):
+            raise TaskOptionsError("api_name must be included in each entry")
+
+        if not all(
+            "internal_sharing_model" in entry or "external_sharing_model" in entry
+            for entry in self.options["org_wide_defaults"]
+        ):
+            raise TaskOptionsError("Each entry must include a sharing model to check.")
+
+        self.owds = {
+            entry["api_name"]: (
+                entry.get("internal_sharing_model"),
+                entry.get("external_sharing_model"),
+            )
+            for entry in self.options["org_wide_defaults"]
+        }
+
+    def _check_owds(self, sobject, result):
+        internal = (
+            result["InternalSharingModel"] == self.owds[sobject][0]
+            if self.owds[sobject][0]
+            else True
+        )
+        external = (
+            result["ExternalSharingModel"] == self.owds[sobject][1]
+            if self.owds[sobject][1]
+            else True
+        )
+        return internal and external
+
+    def _run_task(self):
+        try:
+            ext = (
+                ", ExternalSharingModel"
+                if any(owd[1] is not None for owd in self.owds.values())
+                else ""
+            )
+            self.return_values = all(
+                self._check_owds(
+                    obj,
+                    self.sf.query(
+                        f"SELECT InternalSharingModel{ext} "
+                        "FROM EntityDefinition "
+                        f"WHERE QualifiedApiName = '{obj}'"
+                    )["records"][0],
+                )
+                for obj in self.owds.keys()
+            )
```

Comment:
Can you give me a sense of how this check will be used? Would it make sense for it to return more detailed information instead of just a boolean so that preflight checks can show more specific messages about what is wrong?

Reply:
For target product, we need to check whether the OWDs of two specific Outbound Funds objects are set to Private. If and only if they're set to Private, we can deploy a Sharing Set to customize sharing for the Community, but we won't set the OWDs if the user has configured them otherwise.

I think in many cases it will make more sense to direct the user to a configuration guide section, since there may be a large number of differences between the org shape and the target configuration. Is there a way for the message of a preflight check to include the literal return value from the task?

Reply:
> Is there a way for the message of a preflight check to include the literal return value from the task?

Not currently. And that would probably make it more difficult to handle translations for preflight check messages. (Which, to be fair, already needs work -- I think we are currently extracting them for translation but not actually applying those translations when we display the messages.)